#mean transcript complexity analysis for human and mouse

data_lncRNA <- read.csv("lncrna_main_h.csv", header = F)
colnames(data_lncRNA) <- c('Gene','Gene_length','Transcript','Exon')
data_CDS <- read.csv("mrna_main_h.csv", header = F)
colnames(data_CDS) <- c('Gene','Gene_length','Transcript','Exon')

library(dplyr)
data_lncRNA_new <- mutate(data_lncRNA, TPE = Transcript/Exon)
hist(data_lncRNA_new$TPE)
data_CDS_new <- mutate(data_CDS, TPE = Transcript/Exon)
hist(data_CDS_new$TPE)
mean(data_lncRNA_new$TPE)
mean(data_CDS_new$TPE, na.rm=T)
sd(data_lncRNA_new$TPE)
sd(data_CDS_new$TPE, na.rm=T)
length(data_lncRNA_new$TPE)
length(data_CDS_new$TPE)

data_lncRNA1 <- read.csv("lncrna_main_m.csv", header = F)
colnames(data_lncRNA1) <- c('Gene','Gene_length','Transcript','Exon')
data_CDS1 <- read.csv("mrna_main_m.csv", header = F)
colnames(data_CDS1) <- c('Gene','Gene_length','Transcript','Exon')

data_lncRNA_new1 <- mutate(data_lncRNA1, TPE = Transcript/Exon)
hist(data_lncRNA_new1$TPE)
data_CDS_new1 <- mutate(data_CDS1, TPE = Transcript/Exon)
hist(data_CDS_new1$TPE)
mean(data_lncRNA_new1$TPE)
mean(data_CDS_new1$TPE, na.rm=T)
sd(data_lncRNA_new1$TPE)
sd(data_CDS_new1$TPE, na.rm=T)
length(data_lncRNA_new1$TPE)
length(data_CDS_new1$TPE)

#plotting the distribution curve
h1 <- density(data_CDS_new$TPE, na.rm = T, kernel = "cosine", bw = "bcv", adjust = 3)
h2 <- density(data_lncRNA_new$TPE, kernel = "cosine", bw = "bcv", adjust = 3)
h3 <- density(data_CDS_new1$TPE, na.rm = T, kernel = "cosine", bw = "bcv", adjust = 3)
h4 <- density(data_lncRNA_new1$TPE, kernel = "cosine", bw = "bcv", adjust = 3)
tiff("/home/dell/Documents/koushiki/final_fig/fig_a_b.tiff",width = 2400, height = 1200, res = 300)
par(mar=c(4.1,4.5,1.5,1.5))
par(mfrow=c(1,2))
plot(h1, xlim=c(0,1), col="#E8A735", ylim=c(0,3),xlab="Transcript Complexity", lwd = 4, main = "",cex.lab = 1.3, cex.axis=0.8, font.lab=2, font.axis=2)
lines(h2, col="#4D648D", lwd=4)
plot(h3, xlim=c(0,1), col="#E8A735", ylim=c(0,3),xlab="Transcript Complexity", lwd = 4, main="",cex.lab = 1.3, cex.axis=0.8, font.lab=2, font.axis=2)
lines(h4, col="#4D648D", lwd=4)
dev.off()

#human (boxplot)
data_lncRNA <- read.csv("lncrna_main_h.csv", header = FALSE)
data_lncRNA <- na.omit(data_lncRNA)
names(data_lncRNA)
colnames(data_lncRNA) <- c("Gene","Gene_length","Transcript","Exon")
summary(data_lncRNA)
str(data_lncRNA)
data_lncRNA$Gene_length <- as.numeric(data_lncRNA$Gene_length)
data_lncRNA$Transcript <- as.numeric(data_lncRNA$Transcript)
data_lncRNA$Exon <- as.numeric(data_lncRNA$Exon)
data_lncRNA <- mutate(data_lncRNA, TPE=Transcript/Exon)
median(data_lncRNA$TPE)

data_pcg <- read.csv("mrna_main_h.csv", header = FALSE)
data_pcg <- na.omit(data_pcg)
names(data_pcg)
colnames(data_pcg) <- c("Gene","Gene_length","Transcript","Exon")
summary(data_pcg)
str(data_pcg)
data_pcg$Gene_length <- as.numeric(data_pcg$Gene_length)
data_pcg$Transcript <- as.numeric(data_pcg$Transcript)
data_pcg$Exon <- as.numeric(data_pcg$Exon)
data_pcg <- mutate(data_pcg, TPE=Transcript/Exon)
median(data_pcg$TPE)

tiff("/home/dell/Documents/koushiki/final_fig/fig_a1.tiff",width = 1200, height = 1200, res = 300)
par(mar=c(0.2,5,0.2,0.2))
boxplot(data_lncRNA$TPE, data_pcg$TPE, col = c("#4D648D", "#E8A735"),ylab="TC", boxwex=2.0,
        xaxs="i",at=c(1,4), cex.axis=1.5, font.axix=2,cex.lab=1.8, font.lab=2,
        outline=F, xlim=c(0,5), ylim=c(0.0,0.8), main="")
dev.off()

#mouse (boxplot)
data_lncRNA <- read.csv("lncrna_main_m.csv", header = FALSE)
data_lncRNA <- na.omit(data_lncRNA)
names(data_lncRNA)
colnames(data_lncRNA) <- c("Gene","Gene_length","Transcript","Exon")
summary(data_lncRNA)
str(data_lncRNA)
data_lncRNA$Gene_length <- as.numeric(data_lncRNA$Gene_length)
data_lncRNA$Transcript <- as.numeric(data_lncRNA$Transcript)
data_lncRNA$Exon <- as.numeric(data_lncRNA$Exon)
data_lncRNA <- mutate(data_lncRNA, TPE=Transcript/Exon)
median(data_lncRNA$TPE)

data_pcg <- read.csv("mrna_main_m.csv", header = FALSE)
data_pcg <- na.omit(data_pcg)
names(data_pcg)
colnames(data_pcg) <- c("Gene","Gene_length","Transcript","Exon")
summary(data_pcg)
str(data_pcg)
data_pcg$Gene_length <- as.numeric(data_pcg$Gene_length)
data_pcg$Transcript <- as.numeric(data_pcg$Transcript)
data_pcg$Exon <- as.numeric(data_pcg$Exon)
data_pcg <- mutate(data_pcg, TPE=Transcript/Exon)
median(data_pcg$TPE)

tiff("/home/dell/Documents/koushiki/final_fig/fig_b1.tiff",width = 1200, height = 1200, res = 300)
par(mar=c(0.2,5,0.2,0.2))
boxplot(data_lncRNA$TPE, data_pcg$TPE, col = c("#4D648D", "#E8A735"),ylab="TC", boxwex=2.0,
        xaxs="i",at=c(1,4), cex.axis=1.5, font.axix=2,cex.lab=1.8, font.lab=2,
        outline=F, xlim=c(0,5), ylim=c(0.0,0.8), main="")
dev.off()
